src/event/event.tsx

```diff
@@ -0,0 +1,175 @@
+import BaseSlideExecution from "../base-slide-execution";
+import Logo from "../kk/Logo/Logo";
+import "../kk/Font/font.scss";
+
+/**
+ * Event details component.
+ *
+ * @param {object} props Props.
+ * @param {object} props.date The date.
+ * @param {boolean} props.time The time.
+ * @param {object} props.title The title.
+ * @param {Function} props.subTitle The subtitle.
+ * @returns {object} The component.
+ */
+const EventDetails: FC<EventDetailsProps> = ({
+  date,
+  time,
+  title,
+  subTitle,
+}) => {
+  return (
+    <div className="event-details">
+      <div className="event-details__date">
+        <span>{date}</span>
+        <span>{time}</span>
+      </div>
+      <div className="event-details__title">
+        <h1>{title}</h1>
+      </div>
+      <div className="event-details__sub-title">
+        <h2>{subTitle}</h2>
+      </div>
+    </div>
+  );
+};
+
+/**
+ * Event component.
+ *
+ * @param {object} props Props.
+ * @param {object} props.slide The slide.
+ * @param {boolean} props.run Whether or not the slide should start running.
+ * @param {object} props.content The slide content.
+ * @param {Function} props.slideDone Function to invoke when the slide is done playing.
+ * @returns {object} The component.
+ */
+const Event: FC<EventProps> = ({ slide, run, content, slideDone }) => {
+  const ref = useRef<HTMLDivElement>(null);
+
+  const getDimensions = (): Dimensions => ({
+    width: ref.current?.offsetWidth || 0,
+    height: ref.current?.offsetHeight || 0,
+  });
+
+  const [dimensions, setDimensions] = useState<Dimensions>({
+    width: 0,
+    height: 0,
+  });
+
+  useEffect(() => {
+    const handleResize = () => {
+      setDimensions(getDimensions());
+    };
+
+    if (ref.current) {
+      handleResize();
+    }
+
+    // The resize event will probably never occur in real life, but there's not
+    // reason to not support it.
+    window.addEventListener("resize", handleResize);
+    return () => {
+      window.removeEventListener("resize", handleResize);
+    };
+  }, [ref]);
+
+  // Content from content.
+  const {
+    host,
+    date,
+    time,
+    title,
+    subTitle,
+    image,
+    duration = 15000,
+  } = content;
+
+  const bgColor = content.bgColor || "#000c2e";
+  const textColor = content.textColor || "#fff";
+
+  let layout = "horizontal";
+  if (dimensions.height > dimensions.width) {
+    layout = "vertical";
+  }
+
+  const rootClasses: string[] = [
+    "template-event",
+    "event",
+    "font-kbh",
+    `layout-${layout}`,
+  ];
+  const rootStyle: Record<string, string | number> = {
+    backgroundColor: bgColor,
+    color: textColor,
+  };
+
+  /** Setup slide run function. */
+  const slideExecution = new BaseSlideExecution(slide, slideDone);
+  useEffect(() => {
+    if (run) {
+      slideExecution.start(duration);
+    }
+
+    return function cleanup() {
+      slideExecution.stop();
+    };
+  }, [run]);
+
+  if (layout === "vertical") {
```

Comment:
I find it weird that we're not provided this info somehow from the framework. You're method for determining the layout is solid though.

Should we pull this out into something reusable if we end up with other slides needing the same functionality?

Reply:
My thinking was to make it re-usable when we need to re-use it. Also not sure how we will end up determining layout type for all. As we in general have 3 layouts, landscape, portrait, square but slightly portrait, so aim was to get something done we can experiment with on real screens and see what kind of adjustments are needed.